pkg/functionconfig/types.go

```diff
@@ -96,6 +97,55 @@ type Trigger struct {
 	MaxWorkers int `json:"maxWorkers,omitempty"`
 }
 
+type BatchConfiguration struct {
+	Mode      BatchMode `json:"mode,omitempty"`
+	BatchSize int       `json:"batchSize,omitempty"`
+	Timeout   string    `json:"timeout,omitempty"`
+}
+
+type BatchMode string
+
+const (
+	BatchModeEnable  BatchMode = "enable"
+	BatchModeDisable BatchMode = "disable"
+
+	DefaultBatchSize    = 10
+	DefaultBatchTimeout = "1s"
+)
+
+func BatchModeEnabled(batchConfiguration *BatchConfiguration) bool {
+	if batchConfiguration == nil {
+		return false
+	}
+	return batchConfiguration.Mode == BatchModeEnable
+}
+
+var triggerKindsSupportBatching = []string{
+	"http",
```

Comment:
One day we will replace all of these with constants 🙄 
Not in this PR though

Reply:
@TomerShor list cannot be a constant in golang

Reply:
I was talking about the kinds:
```
    TriggerKindHTTP = "http"
    TriggerKindCron = "cron"
    ...
```

Reply:
@TomerShor gotcha, then yes - we definitely need it :100: